egs/wsj/s5/steps/data/reverberate_data_dir.py

```diff
@@ -0,0 +1,508 @@
+      for item in list:
+          if item.probability is None:
+              item.probability = uniform_probability
+          else:
+              # smooth the probability
+              item.probability = (1 - smoothing_weight) * item.probability + smoothing_weight * uniform_probability
+
+      # Normalize the probability
+      sum_p = sum(item.probability for item in list)
+      for item in list:
+          item.probability = item.probability / sum_p
+
+    return list
+
+# This function creates the RIR list 
+# Each noise item in the list contains the following attributes:
+# rir_id, room_id, receiver_position_id, source_position_id, rt60, drr, probability
+# Please refer to the help messages in the parser for the meaning of these attributes
+def ParseRirList(rir_list_file):
+    rir_parser = argparse.ArgumentParser()
+    rir_parser.add_argument('--rir-id', type=str, required=True, help='This id is unique for each RIR and the noise may associate with a particular RIR by refering to this id')
+    rir_parser.add_argument('--room-id', type=str, required=True, help='This is the room that where the RIR is generated')
+    rir_parser.add_argument('--receiver-position-id', type=str, default=None, help='receiver position id')
+    rir_parser.add_argument('--source-position-id', type=str, default=None, help='source position id')
+    rir_parser.add_argument('--rt60', type=float, default=None, help='RT60 is the time required for reflections of a direct sound to decay 60 dB.')
+    rir_parser.add_argument('--drr', type=float, default=None, help='Direct-to-reverberant-ratio of the impulse.')
+    rir_parser.add_argument('--probability', type=float, default=None, help='probability of the impulse.')
+    rir_parser.add_argument('rir_file_location', type=str, help='rir file location')
+
+    rir_list = []
+    rir_lines = map(lambda x: x.strip(), open(rir_list_file))
+    for line in rir_lines:
+        rir = rir_parser.parse_args(line.split())
+        setattr(rir, "iso_noise_list", [])
+        rir_list.append(rir)
+
+    return SmoothProbabilityDistribution(rir_list)
+
+
+# This function divides the global RIR list into local lists
+# according to the room where the RIRs are generated
+# It returns the room dictionary indexed by the room id
+# Its values are objects with two attributes: a local RIR list
+# and the probability of the corresponding room
+def MakeRoomDict(rir_list):
+    room_dict = {}
+    for rir in rir_list:
+        if rir.room_id not in room_dict:
+            # add new room
+            room_dict[rir.room_id] = lambda: None
+            setattr(room_dict[rir.room_id], "rir_list", [])
+            setattr(room_dict[rir.room_id], "probability", 0)
+        room_dict[rir.room_id].rir_list.append(rir)
+
+    # the probability of the room is the sum of probabilities of its RIR
+    for key in room_dict.keys():
+        room_dict[key].probability = sum(rir.probability for rir in room_dict[key].rir_list)
+
+    return room_dict
```

Comment:
Assert that the probability of rooms sums to 1, as you use this assumption during sampling.